documentation/authentication.md

```diff
@@ -0,0 +1,135 @@
+Authentication
+================
+
+Authenticating requests to `dotnet-monitor` helps protect sensitive diagnostic artifacts from unauthorized users and lower privileged processes. `dotnet-monitor` can be configured to use either [Windows Authentication](#windows-authentication) or via an [API Token](#api-token-authentication). It is possible, although strongly not recommended, to [disable authentication](#disabling-authentication).
+
+## Windows Authentication
+We only recommend using Windows Authentication if you're running `dotnet-monitor` as a local development tool on Windows, for all other environments using an [API Token](#api-token-authentication) is recommended.
+
+Windows authentication doesn't require explicit configuration and is enabled automatically when running `dotnet-monitor` on Windows. When available, dotnet-monitor will authorize any user authenticated as the same user that started the `dotnet-monitor` process. It is not possible to disable Windows authentication.
+
+> **FACT CHECK:** Windows Authentication cannot be disabled when running on Windows. See: [DiagnosticsMonitorCommandHandler](https://github.com/dotnet/dotnet-monitor/blob/69542c44b7d70a83901b93a73d0ea09b90870bcd/src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs#L136)
+
+## API Token Authentication
+An API Token is the recommended authentication mechanism for `dotnet-monitor`. To enable it, you will need to generate a secret key, update the configuration of `dotnet monitor`, and then specify the API Token in the Authorization header on all requests to `dotnet-monitor`.
+
+> **NOTE:** API Token Authentication should only be used when [TLS is enabled](#) to protect the token while in transit.
+
+### Generating an API Token
+
+The API Token you use to secure `dotnet-monitor` should be a 32-bit cryptographically random secret. You can generate a token either using `dotnet-monitor` or via your shell. To generate an API Token with `dotnet-monitor`, simply invoke the `generatekey` command:
+
+```powershell
+dotnet-monitor generatekey
+```
+
+The output from this command will display the API Token formatted as an authentication header along with its hash and associated hashing algorithm. You will need to store the `ApiKeyHash` and `ApiKeyHashType` in the configuration for `dotnet-monitor` and use the authorication header value when making requests to the `dotnet-monitor` HTTPS endpoint.
+
+```
+Authorization: MonitorApiKey Uf6Yq8ZGkcn+ltq2QLHxuXpKA6/Q5VH5Mb5aSIJBxRc=
+ApiKeyHash: CB233C3BE9F650146CFCA81D7AA608E3A3865D7313016DFA02DAF82A2505C683
+ApiKeyHashType: SHA256
+```
+
+> NOTE: The generated API token should be secured at rest. We recommend using a tool such as a password manager to save it.
```

Comment:
Note that other hash algorithms are allowed, but md5 and sha1 are not.

Reply:
Happen to have handy the list of hash algorithms that *are* allowed, @wiktork?

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.hashalgorithm.create?view=net-5.0#System_Security_Cryptography_HashAlgorithm_Create_System_String_

Effectively, anything in that list that is not SHA1 or MD5 (basically SHA256, SHA384, and SHA512).